modules/apps/29-fee/keeper/relay.go

```diff
@@ -16,14 +16,19 @@ func (k Keeper) SendPacket(ctx sdk.Context, chanCap *capabilitytypes.Capability,
 
 // WriteAcknowledgement wraps IBC ChannelKeeper's WriteAcknowledgement function
 // ICS29 WriteAcknowledgement is used for asynchronous acknowledgements
-func (k Keeper) WriteAcknowledgement(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI, acknowledgement []byte) error {
+func (k Keeper) WriteAcknowledgement(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI, acknowledgement ibcexported.Acknowledgement) error {
+	if !k.IsFeeEnabled(ctx, packet.GetDestPort(), packet.GetDestChannel()) {
+		// ics4Wrapper may be core IBC or higher-level middleware
+		return k.ics4Wrapper.WriteAcknowledgement(ctx, chanCap, packet, acknowledgement)
```

Comment:
Any thoughts on the best way to add a test case for this? Setting the fee to disabled is straight forward but not sure what state we can check to assert it was called as expected.

Reply:
Use mock module with async acks, disable fees as you say, then write acknowledgement using mock keeper and check that the acknowledgement gets stored unaltered